### PR TITLE
Spring Boot 3 and JDK 21

### LINKED ISSUE #827: Update dependencies to fix critical and high severity vulnerabilities
### Current Behavior

Hi,

There are dependencies imported with High to Critical severity vulnerabilities. Wouldn't you mind bumping its versions?

- CVE-2022-1471 [CRITICAL], `org.yaml_sankeyaml`@1.30
- CVE-2022-25857 [HIGH], `org.yaml_sankeyaml`@1.30
- CVE-2022-42003 [HIGH], `com.fasterxml.jackson.core_jackson-databind`@2.13.4

### Expected Behavior

Recommended versions:
- `org.yaml_sankeyaml`@1.31
- `com.fasterxml.jackson.core_jackson-databind`@2.14.0

### Steps To Reproduce

_No response_

### Environment

- keycloak-config-cli Version: v5.5.0
- Java Version: 11

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jonasvoelcker**
**What this PR does / why we need it**:

The OSS support for Spring Boot 2 ended in November 2023 and there are plenty of vulnerabilities which might be addressed. This change keeps compatibility to all relevant versions which were supported before and recovered the 19.0.3-legacy version as well.

**Which issue this PR fixes**: fixes #827 #874 #938

**Special notes for your reviewer**:

* Changed Java target version and temurin to 21
* Several dependency updates
* Reassured compatibility with 19.0.3-legacy

***Breaking***
* Upgrade to Spring Boot 3
    * This affects the capability of the path matcher

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Comment by st3v0rr** [comment]
Review looks good. Before we merge it...
We will drop support for any Java version besides Java 21.
Last chance for objections until tonight :-)

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -33,22 +33,22 @@ jobs:
         env:
           # we keep 18.0.2 for backwards compatibility with RH-SSO 7.6
           - KEYCLOAK_VERSION: 18.0.2
-          - KEYCLOAK_VERSION: 21.1.1
-          - KEYCLOAK_VERSION: 22.0.4
+          - KEYCLOAK_VERSION: 21.1.2
+          - KEYCLOAK_VERSION: 22.0.5
```
Comment (st3v0rr) [inline .github/workflows/ci.yaml:37]: @jonasvoelcker since only the tests fail with this 22.0.5 version can you maybe give it a try with the old version?

**Comment by st3v0rr** [comment]
As soon as the issue with keycloak 22 tests is fixed I will merge it and perform a release.

File: src/test/java/de/adorsys/keycloak/config/test/util/SubGroupUtil.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public static List<GroupRepresentation> getSubGroups(
             GroupRepresentation groupRepresentation,
             RealmResource realmResource
     ) {
-        if (CollectionUtils.isNotEmpty(groupRepresentation.getSubGroups())) {
+        if (!CollectionUtils.isEmpty(groupRepresentation.getSubGroups())) {
```
Comment (francis-pouatcha) [inline src/test/java/de/adorsys/keycloak/config/test/util/SubGroupUtil.java:38]: Any rationale behind this change?

File: src/main/java/de/adorsys/keycloak/config/service/ClientPoliciesImportService.java [hunk 3]
```diff
@@ -35,21 +37,33 @@ public class ClientPoliciesImportService {
...
-        logger.trace("Updated client-policy profiles.");
+        if (!StringUtils.isNumeric(keycloakMajor)) {
+            return;
+        }
 
-        ClientPoliciesRepresentation parsedClientPolicies = realmImport.getParsedClientPolicies();
-        clientPoliciesRepository.updateClientPoliciesPolicies(realmImport, parsedClientPolicies);
-        logger.trace("Updated client-policy policies.");
+        if (Integer.parseInt(keycloakMajor) >= 20) {
+            // client-profile profiles must be imported before client-profile policies
```
Comment (francis-pouatcha) [inline src/main/java/de/adorsys/keycloak/config/service/ClientPoliciesImportService.java:59]:
```suggestion
        if (VersionUtil.ge(keycloakProvider.getKeycloakVersion(), "20")) {
            // client-profile profiles must be imported before client-profile policies
```

File: src/main/java/de/adorsys/keycloak/config/service/ClientPoliciesImportService.java [hunk 4]
```diff
@@ -21,7 +21,9 @@
 package de.adorsys.keycloak.config.service;
 
 import de.adorsys.keycloak.config.model.RealmImport;
+import de.adorsys.keycloak.config.provider.KeycloakProvider;
 import de.adorsys.keycloak.config.repository.ClientPoliciesRepository;
+import org.apache.commons.lang3.StringUtils;
```
Comment (francis-pouatcha) [inline src/main/java/de/adorsys/keycloak/config/service/ClientPoliciesImportService.java:26]:
```suggestion
import de.adorsys.keycloak.config.util.VersionUtil;
```

**Review by francis-pouatcha — CHANGES_REQUESTED** [review]

**Pull request merged**